common/lib/xmodule/xmodule/js/src/capa/display.js

```diff
@@ -0,0 +1,1323 @@
+                return $(element).find('input').on('input', function() {
+                    var $p;
+                    $p = $(element).find('span.status');
+                    return $p.parent().removeClass().addClass('unsubmitted');
+                });
+            },
+            choicegroup: function(element) {
+                var $element, id;
+                $element = $(element);
+                id = ($element.attr('id').match(/^inputtype_(.*)$/))[1];
+                return $element.find('input').on('change', function() {
+                    var $status;
+                    $status = $('#status_' + id);
+                    if ($status[0]) {
+                        $status.removeClass().addClass('unanswered');
+                        $status.empty().css('display', 'inline-block');
+                    } else {
+                        $('<span>', {
+                            class: 'unanswered',
+                            style: 'display: inline-block;',
+                            id: 'status_' + id
+                        });
+                    }
+                    return $element.find('label').removeClass();
+                });
+            },
+            'option-input': function(element) {
+                var $select, id;
+                $select = $(element).find('select');
+                id = ($select.attr('id').match(/^input_(.*)$/))[1];
+                return $select.on('change', function() {
+                    return $('#status_' + id).removeClass().addClass('unanswered')
+                        .find('span')
+                        .text(gettext('Status: unsubmitted'));
+                });
+            },
+            textline: function(element) {
+                return $(element).find('input').on('input', function() {
+                    var $p;
+                    $p = $(element).find('span.status');
+                    return $p.parent().removeClass('correct incorrect').addClass('unsubmitted');
+                });
+            }
+        };
+
+        Problem.prototype.inputtypeSetupMethods = {
+            'text-input-dynamath': function(element) {
+                /*
+                 Return: function (eqn) -> eqn that preprocesses the user formula input before
+                 it is fed into MathJax. Return 'false' if no preprocessor specified
+                 */
+                var data, preprocessor, preprocessorClass, preprocessorClassName;
+                data = $(element).find('.text-input-dynamath_data');
+                preprocessorClassName = data.data('preprocessor');
+                preprocessorClass = window[preprocessorClassName];
+                if (preprocessorClass == null) {
+                    return false;
+                } else {
+                    preprocessor = new preprocessorClass();
+                    return preprocessor.fn;
+                }
+            },
+            javascriptinput: function(element) {
```

Comment:
@muzaffaryousaf I'm going to stick to only changing things that cause eslint warnings. There are obviously still many things that could be improved about this code!